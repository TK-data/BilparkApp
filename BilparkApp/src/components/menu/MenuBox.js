import React from 'react';
import { StyleSheet, View, Image, Text, Dimensions, TouchableOpacity } from 'react-native';
import { NavigationActions } from 'react-navigation';
import { connect } from 'react-redux';

const window = Dimensions.get('window');
const styles = StyleSheet.create({
  boxContainer: {
    display: 'flex',
    flexDirection: 'column',
    marginRight: window.width / 40,
    marginLeft: window.width / 40,

  },
  box: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
    // borderStyle: 'solid',
    // borderWidth: 2,
    // borderColor: 'white',
    width: window.width / 4.2,
    height: window.width / 4.2,
    marginLeft: window.width / 60,
    marginRight: window.width / 60,
    marginBottom: window.width / 50,
    borderBottomLeftRadius: 100,
    borderBottomRightRadius: 100,
    borderTopLeftRadius: 100,
    borderTopRightRadius: 100,
  },
  image: {
    width: window.width * 0.15,
    height: window.height * 0.15,
  },
  titleText: {
    color: 'white',
    fontWeight: 'bold',
    textAlign: 'center',
    fontFamily: 'Roboto',
    // marginTop: window.width / 2000,
    marginBottom: window.width / 18,
  },
});

const MenuBox = ({ navigate, image, routeName, pageName }) => {

  return (
    <TouchableOpacity onPress={() =>
      navigate(routeName)}
    >
      <View style={styles.boxContainer}>
        <View style={styles.box}>
          <Image resizeMode="contain" style={styles.image} source={image} />
        </View>
        <Text style={styles.titleText}>{pageName}</Text>
      </View>
    </TouchableOpacity>
  );
};


const mapDispatchToProps = dispatch => ({
  navigate: (routeName) => {
    dispatch(NavigationActions.navigate({ routeName }));
  },
});

const mapStateToProps = () => ({
});

export default connect(mapStateToProps, mapDispatchToProps)(MenuBox);
