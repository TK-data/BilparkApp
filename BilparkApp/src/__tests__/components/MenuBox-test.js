import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import MenuBox from '../../components/menu/MenuBox';

const initialState = {
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing menu boxes', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <MenuBox />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });

  const wrapper = shallow(
    <MenuBox />,
    { context: { store: mockStore(initialState) } },
  );

  it('Pressing the menubox button', () => {
    const render = wrapper.dive();
    console.log(render.debug());
    const submitButton = render.find('TouchableOpacity');
    submitButton.simulate('press');
  });
});
