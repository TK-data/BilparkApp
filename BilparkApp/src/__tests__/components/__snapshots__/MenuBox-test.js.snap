// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing menu boxes Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuBox
    navigate={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "borderRadius": 5,
            "display": "flex",
            "height": 222.33333333333334,
            "justifyContent": "center",
            "margin": 15,
            "width": 214.28571428571428,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={undefined}
          style={
            Object {
              "height": 200.1,
              "width": 112.5,
            }
          }
        />
      </View>,
      "onPress": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Image
          resizeMode="contain"
          source={undefined}
          style={
            Object {
              "height": 200.1,
              "width": 112.5,
            }
          }
        />,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#F5FCFF",
          "borderRadius": 5,
          "display": "flex",
          "height": 222.33333333333334,
          "justifyContent": "center",
          "margin": 15,
          "width": 214.28571428571428,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "resizeMode": "contain",
          "source": undefined,
          "style": Object {
            "height": 200.1,
            "width": 112.5,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5FCFF",
              "borderRadius": 5,
              "display": "flex",
              "height": 222.33333333333334,
              "justifyContent": "center",
              "margin": 15,
              "width": 214.28571428571428,
            }
          }
        >
          <Image
            resizeMode="contain"
            source={undefined}
            style={
              Object {
                "height": 200.1,
                "width": 112.5,
              }
            }
          />
        </View>,
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Image
            resizeMode="contain"
            source={undefined}
            style={
              Object {
                "height": 200.1,
                "width": 112.5,
              }
            }
          />,
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "borderRadius": 5,
            "display": "flex",
            "height": 222.33333333333334,
            "justifyContent": "center",
            "margin": 15,
            "width": 214.28571428571428,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "resizeMode": "contain",
            "source": undefined,
            "style": Object {
              "height": 200.1,
              "width": 112.5,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
